geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AbstractAvroSimpleFeatureTest.scala

```diff
@@ -0,0 +1,93 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.features.avro
+
+import java.io.{File, FileInputStream}
+import java.text.SimpleDateFormat
+import java.util.UUID
+
+import com.vividsolutions.jts.geom.{Point, Polygon}
+import org.geotools.factory.Hints
+import org.geotools.filter.identity.FeatureIdImpl
+import org.locationtech.geomesa.features.avro.serde.Version2ASF
+import org.locationtech.geomesa.utils.geohash.GeohashUtils
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.text.WKTUtils
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Random
+
+trait AbstractAvroSimpleFeatureTest {
+
+  val complexSchema = "f0:String,f1:Integer,f2:Double,f3:Float,f4:Boolean,f5:UUID,f6:Date,f7:Point:srid=4326,"+
+    "f8:Polygon:srid=4326,f9:Long,f10:String,f11:Integer,f12:Date,f13:Geometry,f14:UUID"
+  val complexSft = SimpleFeatureTypes.createType("test", complexSchema)
+
+  def createComplicatedFeatures(numFeatures : Int) : List[Version2ASF] = {
+    val r = new Random()
+    r.setSeed(0)
```

Comment:
It's a minor point, but you could consolidate the previous two lines, initializing the random-number generator with its seed:  ```val r = new Random(0)```